### ISSUE TITLE
Google indexes Template, Helpcenter, and Notion Landing Pages on my Domain

### ISSUE THREAD

**Issue by marcklingen** [stephenou/fruitionsite#63]
Hi @stephenou,

thanks again for your ongoing development and support of this project! While #19 #18 aim to fix this issue, I just would like to reiterate why it would be super important to exclude pages with a 404 or no-index header.

Yesterday, I saw that Google indexed 374 pages on my domain labeled as `Indexed, not submitted in sitemap`. Here's a screenshot with examples:
<img width="485" alt="image" src="https://user-images.githubusercontent.com/2834609/100105035-3162cd80-2e67-11eb-81d4-6308429f855b.png">

While I am not sure about the legal implications, it would be for sure nice to only have own pages in the index.

I am happy to contribute to the solution, let me know how I can help best.

**Comment by stephenou** [comment]
Hey @marcklingen! Yep, I agree we should fix this. The solution I came up with is to 301-redirect unlisted pages back to the homepage.

See https://github.com/stephenou/fruitionsite/commit/7f273edcd4f7ea4707663b3209b9d1eb05aac0bb for what changes you need to make in your script.

Do you mind helping me test it out before I announce it widely? Thanks!

**Comment by marcklingen** [comment]
Thanks @stephenou, the redirect works for me. I'd suggest to also add `X-Robots-Tag: noindex` to the header of the response.

**Comment by marcklingen** [comment]
Just found an exception to the rule in the screenshot. While this solution solves most of the problem, it does not address pages which do not have the characteristic page id, e.g.: `/tools-and-craft/01-andy-hertzfeld`.

**Comment by ThallyssonKlein** [comment]
@marcklingen How did you manage to index the domain on Google? Mine does not appear ...

**Comment by marcklingen** [comment]
@ThallyssonKlein If you domain is not automatically indexed, you can add the `/sitemap.xml` in the [Search Console](https://search.google.com/search-console/about).

**Comment by ThallyssonKlein** [comment]
@marcklingen Where does this `sitemap.xml` file come from?

**Comment by marcklingen** [comment]
@ThallyssonKlein The sitemap is generated by the worker, you can find the line here: https://github.com/stephenou/fruitionsite/blob/7f273edcd4f7ea4707663b3209b9d1eb05aac0bb/worker.js#L91

**Comment by lasharor** [comment]
Had the same problem. Changed the code as per 7f273ed. Pages indexed by google now all go back to the homepage.

**Comment by marcklingen** [comment]
@lasharor That does not work for pages with a nice slug such as `/pricing`

**Comment by ThallyssonKlein** [comment]
![image](https://user-images.githubusercontent.com/16887598/102393332-f9a6fb80-3fb6-11eb-8485-a857f45b560a.png)

@marcklingen How long can indexing take?

**Comment by marcklingen** [comment]
@ThallyssonKlein Usually it takes a couple of days but less than a week.

**Comment by ThallyssonKlein** [comment]
@marcklingen I have been notified that the pages are not being read correctly

![image](https://user-images.githubusercontent.com/16887598/102919014-ceb12180-4466-11eb-8f72-84baf987a2cd.png)

**Comment by stephenou** [comment]
> Just found an exception to the rule in the screenshot. While this solution solves most of the problem, it does not address pages which do not have the characteristic page id, e.g.: `/tools-and-craft/01-andy-hertzfeld`, `/pricing`

Hey Marc, yeah unfortunately that's true. One solution is to define a denylist of URLs that Notion uses for marketing, but it's hard to keep it updated when Notion adds new pages. Another solution is to define an allowlist of URLs that your site can visit, but it's also hard to keep it updated when you add new pages.

**Comment by shyjal** [comment]
Try https://technicalseo.com/tools/fetch-render/ to fetch your site as the google bot.

You can see Notion is blocking google and bing's user agent. 404 is not something fruition is doing. Maybe extra code can bypass this

**Comment by vlafriday** [comment]
> Hi @stephenou,
> 
> thanks again for your ongoing development and support of this project! While #19 #18 aim to fix this issue, I just would like to reiterate why it would be super important to exclude all pages that are not whitelisted with a 404 or no-index header.
> 
> Yesterday, I saw that Google indexed 374 pages on my domain labeled as `Indexed, not submitted in sitemap`. Here's a screenshot with examples: <img alt="image" width="485" src="https://user-images.githubusercontent.com/2834609/100105035-3162cd80-2e67-11eb-81d4-6308429f855b.png">
> 
> While I am not sure about the legal implications, it would be for sure nice to only have own pages in the index.
> 
> I am happy to contribute to the solution, let me know how I can help best.

Hi, @marcklingen, I know it's been many years, but maybe you know the solution? Now I get the same problem with indexing. Although sitemap.xml has been added to Google Console.

I see your website (https://marcklingen.com/) it works without problems. How did you do that?
Can you share and help?

![Screenshot_35](https://user-images.githubusercontent.com/78411322/235828422-52f77b44-42cf-41a8-bd22-3a9b78340558.png)
![Screenshot_36](https://user-images.githubusercontent.com/78411322/235828430-4c5fdd86-1997-4178-b41c-7e0daa6d6ebe.png)

@stephenou

**Comment by marcklingen** [comment]
Hi @vlafriday, I switched to [react-notion-x](https://github.com/NotionX/react-notion-x) for this page while still using fruition for others. Based on the screenshot (which I do not fully understand) your problem looks different to the one I had as you have a 404 and not too many pages in the index.

**Comment by vlafriday** [comment]
Do you have good indexing in google console? Does it work? Are you using Vercel to deploy code?

> Hi @vlafriday, I switched to [react-notion-x](https://github.com/NotionX/react-notion-x) for this page while still using fruition for others. Based on the screenshot (which I do not fully understand) your problem looks different to the one I had as you have a 404 and not too many pages in the index.

**Comment by marcklingen** [comment]
> Do you have good indexing in google console? Does it work? Are you using Vercel to deploy code?
> 
> > Hi @vlafriday, I switched to [react-notion-x](https://github.com/NotionX/react-notion-x) for this page while still using fruition for others. Based on the screenshot (which I do not fully understand) your problem looks different to the one I had as you have a 404 and not too many pages in the index.

Yes, works well for me and I deploy on Vercel. You can also go for managed solutions like [super.so](https://super.so), have not tried them though.
